File: backend/app/api/endpoints/ppt.py
from fastapi import APIRouter, Depends
from fastapi.responses import FileResponse

from backend.app.core.security import verify_access_token
from backend.app.models.contracts import PPTRequest, PPTResult
from backend.app.services.ppt import generate, resolve_download

router = APIRouter(dependencies=[Depends(verify_access_token)])

@router.post("/generate", response_model=PPTResult)
def generate_ppt(request: PPTRequest): return generate(request)

@router.get("/download/{file_id}")
def download_ppt(file_id: str):
    path, display_name = resolve_download(file_id); return FileResponse(path, filename=display_name, media_type="application/vnd.openxmlformats-officedocument.presentationml.presentation")

File: frontend/src/api/ppt.ts
import { api } from './client'
import type { PPTRequest, PPTResult } from '../types/api'
export const generatePpt = async (input:PPTRequest) => (await api.post<PPTResult>('/ppt/generate',input)).data
export const getPptDownloadUrl = (fileId:string) => `${api.defaults.baseURL ?? '/api'}/ppt/download/${encodeURIComponent(fileId)}`

